### ISSUE TITLE
[logistic-regression.ipynb] is self.weights updated ?

### ISSUE THREAD

**Issue by y-d-y** [rasbt/deeplearning-models#80]
FilePath:
deeplearning-models/pytorch_ipynb/basic-ml/logistic-regression.ipynb    

def train(self, x, y, num_epochs, learning_rate=0.01):
        
        for e in range(num_epochs):
            
            #### Compute outputs ####
            proba = self.forward(x)
            cost = self._logit_cost(y, proba)
            
            #### Compute gradients ####
            cost.backward()
            
            #### Update weights ####
            
            tmp = self.weights.detach()
            tmp -= learning_rate * self.weights.grad
            
            tmp = self.bias.detach()
            tmp -= learning_rate * self.bias.grad
            
            #### Reset gradients to zero for next iteration ####
            self.weights.grad.zero_()
            self.bias.grad.zero_()
    
            #### Logging ####
            print('Epoch: %03d' % (e+1), end="")
            print(' | Train ACC: %.3f' % self.evaluate(x, y), end="")
            print(' | Cost: %.3f' % self._logit_cost(y, self.forward(x)))

tmp did not update self.weights

**Comment by y-d-y** [comment]
sorry!  tmp and self.weights share memory
